File: Interviewbit/Heap And Maps/profit-maximisation.java
public class Solution {
    public int solve(ArrayList<Integer> A, int B) {
        Queue<Integer>pq=new PriorityQueue<>(Collections.reverseOrder());
        pq.addAll(A);
        int ans=0,tp=0;
        while((B--)!=0){
            tp=pq.poll();
            ans+=tp;
            if((--tp)!=0) pq.add(tp);
        }
        return ans;
    }
}

File: Interviewbit/Heap And Maps/profit-maximisation.cpp
int Solution::solve(vector<int> &A, int B) {
    
    priority_queue<int>pq;
    for(int i=0;i<A.size();i++)
        pq.push(A[i]);
        
    int ans=0;
    int tp=0;
    while(B--){
        tp=pq.top();
        pq.pop();
        ans+=tp;
        if(--tp)pq.push(tp);
    }
    return ans;
}
